walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,93 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.Collection;
 
+
 public final class FindMeetingQuery {
+
+  private Collection<TimeRange> collect = new ArrayList<TimeRange>();
+  private boolean flag = true;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    long duration = request.getDuration();                 // Duration of the meeting, request
+    Collection<String> attendees = request.getAttendees(); // Attendees of the meeting, request
+
+    // For any request, where the duration is greater than 24 hours. No 
+    //  times can be given.
+    if (duration > TimeRange.WHOLE_DAY.duration()){
+        return collect;
+    }
+    // For any request, where no events are schedule. The whole day can be
+    //  used.
+    else if(events.isEmpty()){ 
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    collect.add(TimeRange.WHOLE_DAY);
```

Comment:
Is collect meant to signify available times or unavailable times for scheduling the event?